src/lib.rs

```diff
@@ -733,6 +797,51 @@ impl Certificate {
     pub fn stake_pool_registration(pool_registration: PoolRegistration) -> Certificate {
         certificate::Certificate::PoolRegistration(pool_registration.0).into()
     }
+
+    /// Create a Certificate for PoolRetirement
+    pub fn stake_pool_retirement(pool_retirement: PoolRetirement) -> Certificate {
```

Comment:
Prefer to always clone function parameters (`&PoolRetirement`) otherwise it can lead to unexpected null pointer errors in Javascript